File: pesocial-frontend/src/services/creatorService.ts
import api from './api'
import type { CreatorAnalytics } from '../types/analytics'

export const creatorService = {
  getAnalytics: async (creatorId: string) => {
    const { data } = await api.get<CreatorAnalytics>(`/api/creators/${creatorId}/analytics`)
    return data
  },
}

File: src/main/java/com/pesocial/service/CreatorService.java
package com.pesocial.service;

import com.pesocial.model.analytics.CreatorAnalytics;
import com.pesocial.model.post.Post;

public interface CreatorService {
    Post uploadPost(Post post);
    Post uploadExclusivePost(Post post);
    void startLiveStream();
    boolean enableMonetization(String creatorId);
    CreatorAnalytics viewAnalytics(String creatorId);
    String generateAnalyticsReport(String creatorId);
    void pinPost(String postId);
}
